File: poc/mobile/poc_flutter/lib/features/auth/data/auth_api.dart
import 'package:dio/dio.dart';
import 'models/user_model.dart';

import 'package:dio/dio.dart';
import 'models/user_model.dart';

class AuthApi {
  final Dio dio;
  AuthApi(this.dio);

  Future<UserModel> register(String email, String password) async {
    final res = await dio.post(
      '/register',
      data: {"email": email, "password": password},
      options: Options(headers: {"Content-Type": "application/json"}),
    );
    return UserModel.fromJson(res.data);
  }

  Future<(String, String)> login(String email, String password) async {
    final res = await dio.post(
      '/auth',
      data: {"email": email, "password": password},
      options: Options(headers: {"Content-Type": "application/json"}),
    );
    final access = res.data['access_token'] as String;
    final refresh = res.data['refresh_token'] as String;
    return (access, refresh);
  }
}

File: poc/mobile/poc_kotlin/app/src/main/java/com/example/poc_kotlin/data/api/AuthApi.kt
package com.example.poc_kotlin.data.api
import com.example.poc_kotlin.data.model.ApiResponse
import com.example.poc_kotlin.data.model.AuthResponse
import com.example.poc_kotlin.data.model.User
import retrofit2.http.Body
import retrofit2.http.POST

data class AuthRequest(
    val email: String,
    val password: String
)

interface AuthApi {
    @POST("register")
    suspend fun register(@Body req: AuthRequest): User

    @POST("auth")
    suspend fun login(@Body req: AuthRequest): AuthResponse
}